openff/models/types.py

```diff
@@ -56,17 +56,22 @@ def validate_type(cls, val):
                     # some custom behavior could go here
                     assert unit_.dimensionality == val.dimensionality
                     # return through converting to some intended default units (taken from the class)
+                    val = float(val.m) * val.u
                     return val.to(unit_)
                     # could return here, without converting
                     # (could be inconsistent with data model - heteregenous but compatible units)
                     # return val
                 if _is_openmm_quantity(val):
                     return _from_omm_quantity(val).to(unit_)
-                if isinstance(val, (float, int)) and not isinstance(val, bool):
+                if isinstance(val, int) and not isinstance(val, bool):
+                    # coerce ints into floats for a FloatQuantity
+                    return float(val) * unit_
+                if isinstance(val, float):
                     return val * unit_
                 if isinstance(val, str):
                     # could do custom deserialization here?
-                    return unit.Quantity(val).to(unit_)
+                    val = unit.Quantity(val).to(unit_)
+                    return float(val.m) * val.u
```

Comment:
It'd be nice to avoid double-creating `Quantity`, but I can't find a way to do that just looking at Pint's source and I sure don't want to pre-process the string. Any ideas? If not, that's fine.